src/librustc_mir/const_eval/machine.rs

```diff
@@ -51,7 +51,7 @@ impl<'mir, 'tcx> InterpCx<'mir, 'tcx, CompileTimeInterpreter<'mir, 'tcx>> {
 
         let gid = GlobalId { instance, promoted: None };
 
-        let place = self.const_eval_raw(gid)?;
+        let place = self.const_eval(gid)?;
```

Comment:
Again this seems surprising -- we don't want valtree conversion here, do we? And it's not codegen with its special needs either. So why not a raw const?

Reply:
the `InterpCx::const_eval` function is what used to be `InterpCx::const_eval_raw`. The old `const_eval` does not exist anymore.

Reply:
The rest of the PR still calls `const_eval_raw` in a few places, and the commit message is 

>  Do not call the `const_eval` query in mir interpretation except for caching of nulary intrinsics

Am I the only one who is very confused now?^^

Reply:
I am not confused about *what* the PR is doing, but I may have completely messed up what it *should* be doing (along with any explanations). Basically I think my mistake was to rename `InterpCx::const_eval_raw` without also renaming `query const_eval_raw`

Reply:
how about now?

Reply:
Ah I think I see it now. :)